False-positive: New invalid-name handling does not handle `elif` correctly

### Bug description

The patch notes for 4.0.0 show that `invalid-name` now tries to figure out if a module-level variable is constant or not. It shows that the following would be treated as a variable:


```python
LIMIT = 500  # [invalid-name]
if sometimes:
    LIMIT = 1  # [invalid-name]
```

The release notes state (correctly) that the following should prevent the constant from being treated as a variable:

```python
if sometimes:
    LIMIT = 1
else:
    LIMIT = 500  # exclusive assignment: uses const regex, no warning
```

However, the following gives a false-positive `invalid-name` alert:

```python
if sometimes == 1:
    LIMIT = 1  # [invalid-name]
elif sometimes == 2:
    LIMIT = 2  # [invalid-name]
else:
    LIMIT = 500  # [invalid-name]
```

### Configuration

```ini

```

### Command used

```shell
python3 -m pylint test3.py
```

### Pylint output

```python
test3.txt:6:3: C0103: Variable name "LIMIT" doesn't conform to snake_case naming style (invalid-name)
test3.txt:8:3: C0103: Variable name "LIMIT" doesn't conform to snake_case naming style (invalid-name)
test3.txt:10:3: C0103: Variable name "LIMIT" doesn't conform to snake_case naming style (invalid-name)
```

### Expected behavior

`LIMIT` is treated as a constant and no `invalid-name` is raised.

### Pylint version

```shell
pylint 4.0.0
astroid 4.0.1
Python 3.12.3 (main, Aug 14 2025, 17:47:21) [GCC 13.3.0]
```

### OS / Environment

Ubuntu 24.04

### Additional dependencies

```python

```

--- a/doc/whatsnew/4/4.0/index.rst
+++ b/doc/whatsnew/4/4.0/index.rst
@@ -74,6 +74,30 @@ to your liking.
 
 .. towncrier release notes start
 
+What's new in Pylint 4.0.2?
+--------------------------------
+Release date: 2025-10-20
+
+
+False Positives Fixed
+---------------------
+
+- Fix false positive for :ref:`invalid-name` on a partially uninferable module-level constant.
+
+  Closes #10652 (`#10652 <https://github.com/pylint-dev/pylint/issues/10652>`_)
+
+- Fix a false positive for :ref:`invalid-name` on exclusive module-level assignments
+  composed of three or more branches. We won't raise :ref:`disallowed-name` on module-level names that can't be inferred
+  until a further refactor to remove this false negative is done.
+
+  Closes #10664 (`#10664 <https://github.com/pylint-dev/pylint/issues/10664>`_)
+
+- Fix false positive for :ref:`invalid-name` for ``TypedDict`` instances.
+
+  Closes #10672 (`#10672 <https://github.com/pylint-dev/pylint/issues/10672>`_)
+
+
+
 What's new in Pylint 4.0.1?
 ---------------------------
 Release date: 2025-10-14

--- a/doc/whatsnew/fragments/10652.false_positive
+++ b/doc/whatsnew/fragments/10652.false_positive
@@ -1,3 +1,0 @@
-Fix false positive for ``invalid-name`` on a partially uninferable module-level constant.
-
-Closes #10652

--- a/doc/whatsnew/fragments/10664.false_positive
+++ b/doc/whatsnew/fragments/10664.false_positive
@@ -1,5 +1,0 @@
-Fix a false positive for ``invalid-name`` on exclusive module-level assignments
-composed of three or more branches. We won't raise ``disallowed-name`` on module-level names that can't be inferred
-until a further refactor to remove this false negative is done.
-
-Closes #10664

--- a/doc/whatsnew/fragments/10672.false_positive
+++ b/doc/whatsnew/fragments/10672.false_positive
@@ -1,3 +1,0 @@
-Fix false positive for ``invalid-name`` for ``TypedDict`` instances.
-
-Closes #10672

--- a/pylint/__pkginfo__.py
+++ b/pylint/__pkginfo__.py
@@ -9,7 +9,7 @@ It's updated via tbump, do not modify.
 
 from __future__ import annotations
 
-__version__ = "4.0.1"
+__version__ = "4.0.2"
 
 
 def get_numversion_from_version(v: str) -> tuple[int, int, int]:

--- a/tbump.toml
+++ b/tbump.toml
@@ -1,7 +1,7 @@
 github_url = "https://github.com/pylint-dev/pylint"
 
 [version]
-current = "4.0.1"
+current = "4.0.2"
 regex = '''
 ^(?P<major>0|[1-9]\d*)
 \.

